base/task.jl

```diff
@@ -767,7 +767,7 @@ end
 
 function enq_work(t::Task)
     (t._state === task_state_runnable && t.queue === nothing) || error("schedule: Task not runnable")
-    if t.sticky || Threads.threadpoolsize() == 1
+    if t.sticky || Threads.maxthreadid() == 1
```

Comment:
```suggestion
    if !t.sticky
        tpid = max(threadpoolid(t), 1)
        if tpid == 1 && Threads.threadpoolsize(tpid) == 1
            ccall(:jl_set_task_tid, Cint, (Any, Cint), t, 1)
            t.sticky = true
        end
    end
    if t.sticky
```
I think this was trying to check specifically the size of the first threadpool, and check if `t` is meant to run in the first threadpool, and if both are 1, then to add it to the sticky queue for the first threadpool.

Reply:
I reworked this to check for the size of the task's preferred thread pool, whichever it is.